src/core/context.rs

```diff
@@ -0,0 +1,26 @@
+use crate::core::types::VariableFactory;
+use crate::jet::Application;
+use std::marker::PhantomData;
+
+/// Context for constructing a Simplicity program
+#[allow(dead_code)]
+pub struct Context<App: Application> {
+    pub(crate) naming: VariableFactory,
```

Comment:
In bdfd1769a4f02a3ba4377868c70d0248beb699a7:

I know this is only a `pub(crate)` but ideally it should be private and there should just be an accessor method for allocating a new variable.

In future I'd like to replace this with a boxed closure actually and support users replacing it with something different; so we don't want to expose too much of the implementation in the API.